src/gazebo_ros_realsense.cpp

```diff
@@ -49,22 +49,36 @@ void GazeboRosRealsense::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
   this->camera_info_manager_.reset(
     new camera_info_manager::CameraInfoManager(this->node_.get(), this->GetHandle()));
 
-  this->itnode_.reset(new image_transport::ImageTransport(this->node_));
+  itnode_.reset(new ImageTransportWithSpecifiedQos(node_));
 
-  this->color_pub_ = this->itnode_->advertiseCamera(
-    cameraParamsMap_[COLOR_CAMERA_NAME].topic_name, 2);
-  this->ir1_pub_ = this->itnode_->advertiseCamera(
-    cameraParamsMap_[IRED1_CAMERA_NAME].topic_name, 2);
-  this->ir2_pub_ = this->itnode_->advertiseCamera(
-    cameraParamsMap_[IRED2_CAMERA_NAME].topic_name, 2);
-  this->depth_pub_ = this->itnode_->advertiseCamera(
-    cameraParamsMap_[DEPTH_CAMERA_NAME].topic_name, 2);
+  itnode_->specify_color_qos(color_pub_,cameraParamsMap_[COLOR_CAMERA_NAME].topic_name,colorQos);
+  itnode_->specify_color_qos(ir1_pub_,cameraParamsMap_[IRED1_CAMERA_NAME].topic_name,colorQos);
+  itnode_->specify_color_qos(ir2_pub_,cameraParamsMap_[IRED2_CAMERA_NAME].topic_name,colorQos);
+  itnode_->specify_color_qos(depth_pub_,cameraParamsMap_[DEPTH_CAMERA_NAME].topic_name,colorQos);
 
   if (pointCloud_) {
-    this->pointcloud_pub_ = this->node_->create_publisher<sensor_msgs::msg::PointCloud2>(
-      pointCloudTopic_, rclcpp::SystemDefaultsQoS());
+      if(pointCloudQos=="SensorDataQoS") {
+        pointcloud_pub_ = node_->create_publisher<sensor_msgs::msg::PointCloud2>(pointCloudTopic_,
+                                                                                   rclcpp::SensorDataQoS());
+        RCLCPP_INFO(node_->get_logger(), "Gazebo ROS Realsense plugin -> publisher created using SensorDataQoS.");
+        } else if(pointCloudQos=="ParametersQoS") {
+            pointcloud_pub_ = node_->create_publisher<sensor_msgs::msg::PointCloud2>(
+                    pointCloudTopic_, rclcpp::ParametersQoS());
+            RCLCPP_INFO(node_->get_logger(), "Gazebo ROS Realsense plugin -> publisher created using ParametersQoS.");
+        } else if(pointCloudQos=="ServicesQoS") {
+            pointcloud_pub_ = node_->create_publisher<sensor_msgs::msg::PointCloud2>(
+                    pointCloudTopic_, rclcpp::ServicesQoS());
+            RCLCPP_INFO(node_->get_logger(), "Gazebo ROS Realsense plugin -> publisher created using ServicesQoS.");
+        } else if(pointCloudQos=="ParameterEventsQoS") {
+            pointcloud_pub_ = node_->create_publisher<sensor_msgs::msg::PointCloud2>(
+                    pointCloudTopic_, rclcpp::ParameterEventsQoS());
+            RCLCPP_INFO(node_->get_logger(), "Gazebo ROS Realsense plugin -> publisher created using ParameterEventsQoS.");
+        } else {
+            pointcloud_pub_ = node_->create_publisher<sensor_msgs::msg::PointCloud2>(
+                    pointCloudTopic_, rclcpp::SystemDefaultsQoS());
+            RCLCPP_INFO(node_->get_logger(), "Gazebo ROS Realsense plugin -> publisher created using SystemDefaultsQoS.");
+        }
```

Comment:
to reduce code you could just have a variable for the QOS in the if else statements and at the end create the publisher

Reply:
ok, will do